backend/infrahub/graphql/resolvers/ipam.py

```diff
@@ -231,38 +232,58 @@ async def _resolve_available_prefix_nodes(
     return available_nodes
 
 
+def _filter_kinds(nodes: list[Node], kinds: list[str], limit: int | None) -> list[Node]:
+    filtered: list[Node] = []
+    available_node_kinds = [InfrahubKind.IPPREFIXAVAILABLE, InfrahubKind.IPRANGEAVAILABLE]
+    kinds_with_available = kinds + available_node_kinds
+
+    limit_with_available = limit
+    for node in nodes:
+        if node.get_schema().kind not in kinds_with_available:
+            continue
+        # Adapt the limit of nodes to return by always including available ones
```

Comment:
Is it ok that we would return more nodes than expected in that case?

Reply:
This is expected. We decided, when working on a previous PR, that the limit should be honoured for existing nodes, and that the ones we create on the fly (in-memory for a query) should not count into the pagination limit.